### DISCUSSION TITLE
Conteggio dei campi

### DISCUSSION THREAD

**Discussion by Androm82** [opendatasicilia/HfcQGIS-md#50] [Domande] [answered]
Nel campo "etat_equip" mi serve avere il conteggio (se presenti) di tutte gli altri campi

es. nella particella 510 c'é un solo elemento (Accés) e il risultato del conteggio é 1

diversamente nella particella 5047 il risultato é 7 perché gli elementi sono presenti quindi la somma é 7

ti ringrazio in anticipo per la tua risposta

![272369691_287650206767156_7639605721686448800_n](https://user-images.githubusercontent.com/98805763/151937420-9afaf7d1-144e-49b4-a70b-a30c3c0f8176.jpg)

**Comment by pigreco** [comment, marked as answer]
Buongiorno @Androm82,

Una soluzione passa dall'usare le [Maps](https://hfcqgis.opendatasicilia.it/gr_funzioni/maps/maps_unico/) e gli [Array](https://hfcqgis.opendatasicilia.it/gr_funzioni/array/array_unico/), quindi aggiornerei l'attributo `etat_equip` con l'espressione:

```
with_variable('deleteField',
        map_delete(map_delete(
        attributes(),'parz_numme'),'etat_equip'),
-- conta elementi diversi da NULL
        array_length(
                  array_filter(
                            map_avals(@deleteField),@element IS NOT NULL)
))
```

![image](https://user-images.githubusercontent.com/7631137/151944560-895a4c0c-51f9-49c9-8476-11b077a3817d.png)


NB: occorre eliminare dal conteggio tutti i campi che non ci interessano, quindi occorre inizialmente (map_delete) eliminare il campo 'parz_numme' e 'etat_equip' (se esistesse già)


[qui ](https://hfcqgis.opendatasicilia.it/esempi/maxValoreCampoNome/)altri esempi più complessi.

**Reply by Androm82** [reply]
grazie mille.
é perfetto

**Comment by pigreco** [comment]
Per chi volesse fare dei test, sotto il file utilizzato
[esempio.csv](https://github.com/opendatasicilia/HfcQGIS-md/files/7977798/esempio.csv)
